src/main/java/com/zendesk/maxwell/MaxwellReplicator.java

```diff
@@ -294,19 +298,22 @@ public RowMap getRow() throws Exception {
 					break;
 				case MySQLConstants.TABLE_MAP_EVENT:
 					tableCache.processEvent(getSchema(), this.filter, (TableMapEvent) v4Event);
+					setReplicatorPosition((AbstractBinlogEventV4) v4Event);
 					break;
 				case MySQLConstants.QUERY_EVENT:
 					QueryEvent qe = (QueryEvent) v4Event;
 					if (qe.getSql().toString().equals("BEGIN"))
 						rowBuffer = getTransactionRows();
```

Comment:
looks like there are some `case` conditions in `getTransactionRows` that also need to save the replicator position? (TABLE_MAP_EVENT, QUERY_EVENT, XID_EVENT)